Cent os5 windows no side by side

mingwandroid opened the pull request:
@msarahan @ccordoba12 this superceeds Carlos' PRs since I rebased my CentOS5 changes ontop of his work.

The builds are in progress but please review what you can (if you can). Time is short on the Qt5 build-out.

mingwandroid committed:
Run the obvci batch file (DO NOT SUBMIT THIS UPSTREAM)

ccordoba12 committed:
Don't try to install this package next to the Qt4 one and simply call it qt

- Also simplify installation options

mingwandroid committed:
qt5: Fix building on CentOS 5.11

ccordoba12 committed:
Fix installation locations on Windows

ccordoba12 committed:
Fix patch_prefix_files.py after changing Qt locations on Windows

ccordoba12 committed:
Add a script to change qt.conf contents per environment

- This is the same one we use for qt4

ccordoba12 committed:
Don't remove shared data on Windows

- It's very important to have it on Qt5

ccordoba12 committed:
Compile Webkit only for VS 2008 and 2010

ccordoba12 committed:
Move prefix to LIBRARY_PREFIX to avoid errors while compiling other Qt packages

ccordoba12 committed:
Change prefix to be $PREFIX to avoid errors compiling other Qt packages

ccordoba12 commented on recipes/qt5/build.sh:
```diff
@@ -1,11 +1,16 @@
+patch -p0 < ${RECIPE_DIR}/0003-qtwebkit-O_CLOEXEC-workaround.patch
+patch -p0 < ${RECIPE_DIR}/0004-qtwebkit-CentOS5-Fix-fucomip-compat-with-gas-2.17.50.patch
 
+rm qtwebkit.tar
```
@mingwandroid, this whole WebKit block needs to be executed only on Linux. We're going to provide WebEngine in OS X.

mingwandroid replied:
OK. I'll update this when my builds finish.

ccordoba12 replied:
Ok

msarahan commented:
Guys, I don't have time to be playing with this this weekend.  Rather than waiting on me to merge this, you guys should submit a separate PR with your fork and then I'll close my original Qt5 PR.

ccordoba12 commented:
Mike, could you merge your current PR (after merging this one) in staged-recipes, so we can keep working on the feedstock?

I think it'd be easier for us :-)

mingwandroid commented:
I'll give @ccordoba12 commit access to my staged-recipes, it'll be fine I think. Sorry to bother, and enjoy the 4th (if you're doing anything).

ccordoba12 committed:
Download Webkit only on Linux

mingwandroid committed:
Disable Ubuntu scrollbar overlays

ccordoba12 committed:
Use $PREFIX for datadir and archdatadir

- This avoids compilation problems with PyQt5

mingwandroid committed:
[WIP] Added comments about gtk being too old

ccordoba12 committed:
Change datadir and archdatadir to %LIBRARY_PREFIX%

ccordoba12 committed:
Use gcc for testing only on Linux

mingwandroid committed:
[WIP]: Add comments about two bugs, GLX and Webkit

mingwandroid committed:
Fix EnableDelayedExpansion

mingwandroid committed:
Fix webengine build failures

Patches gotten directly from Chromium

mingwandroid committed:
Fix Windows deps. Need msinttypes and icu 57.*

mingwandroid committed:
Fix patch_prefix_files.py

It wasn't patching enough and used the wrong dir for LIBQT5

mingwandroid committed:
Fix Windows configure -libdir

It was specified twice and the 2nd time it was incorrect.

mingwandroid committed:
Disable QtWebEngine, tidy up bld.bat, add notes

 * Add `AVOID_WEBENGINE`, set to `yes`.
 * Add and/or improve comments.
 * Use parens instead of `GOTO`.
 * Use :: for comments.
 * Use lower-case for built-ins.
 * Use `pushd` and `popd` instead of `cd`.
 * Updated notes.md to reflect latest state.

Carlos had some success with a VirtualBox VM,
QT_OPENGL=software and (I guess) MSYS2 Mesas
opengl32sw.dll but unfortunately I could not
get QtWebEngine not to crash using these. So
we must disable it.

mingwandroid committed:
Fix paths with spaces on Windows

ccordoba12 commented on recipes/qt5/bld.bat:
```diff
@@ -1,104 +1,137 @@
+) else (
+  set TARGET_ARCH=x86
+)
+call %RECIPE_DIR%\obvci_appveyor_python_build_env.cmd
```
Is there something that let us detect that we are running in conda-forge?

mingwandroid replied:
I am not sure, we could talk about integrating `obvci` into `conda-build` somehow, possibly? @pelson and @msarahan, I would welcome your input.

ccordoba12 commented on recipes/qt5/bld.bat:
```diff
@@ -1,104 +1,137 @@
+  set WEBTHING=qtwebkit
+)
+if "%AVOID_WEBENGINE%" == "yes" (
+  set WEBTHING=qtwebkit
```
Could you change the name of this variable `WEBTHING` for `WEBBACKEND`?

mingwandroid replied:
I actually built a Qt5 which used both by mistake using a WIP version of this file, so I don't think we want to be mutex here finally, what do you think? It would be nice if we could be dynamic with it in the future, so I'll probably make two vars, `USE_WEBKIT` and `USE_WEBENGINE`.

ccordoba12 replied:
Ok, I think it's better

ccordoba12 commented on recipes/qt5/bld.bat:
```diff
@@ -1,104 +1,137 @@
-
-:: Install a custom python 27 environment for us, to use in building, but avoid feature activation
-conda create -y -n python27_qt5_build python=2.7
+set "PATH=%CD%\qtbase\bin;%CD%\gnuwin32\bin;C:\Ruby23\bin;%PATH%"
```
- Could you mention in `Notes.md` that you need to install Ruby before running this?
- Why `gnuwin32` is needed here? Don't you have all that you need from the MSYS2 channel?

mingwandroid replied:
Yes, in theory MSYS2 provides everything we need, but I took a safe / chicken approach here to save time incase there was any problems. I'll revisit this once I get some time to breathe. Note, MSYS2 can also provide `ruby`, but I need to add that to the `msys2` channel which I'll do when I create the next epoch of the `msys2` suite.

ccordoba12 replied:
Ok

ccordoba12 commented on recipes/qt5/bld.bat:
```diff
@@ -1,104 +1,137 @@
-conda create -y -n python27_qt5_build python=2.7
+set "PATH=%CD%\qtbase\bin;%CD%\gnuwin32\bin;C:\Ruby23\bin;%PATH%"
+
+:: Install a custom python 27 environment for us, to use in building webengine, but avoid feature activation
```
So WebEngine can only use Python 2.7 to be built, right? Could you clarify that here?

mingwandroid replied:
Sure, in `notes.md` or directly here?

mingwandroid committed:
Remove TEXTRELs from Linux/x86 qtWebkit

There are refs to absolute symbols in the Webkit JIT
which cause failure to load on SELinux:

ImportError: ../miniconda/envs/test-spy/lib/python2.7/site-packages/PyQt5/../../../libQt5WebKit.so.5: cannot restore segment prot after reloc: Permission denied

mingwandroid committed:
Add disabled patch for gtk-2 support

It needs more work, and we are not sure if
it is essential or not.

mingwandroid committed:
Backport a fix for xcb-glx detection

To fix:
QXcbIntegration: Cannot create platform OpenGL context, neither GLX nor EGL are enabled

mingwandroid commented:
I made most of these changes now @ccordoba12 , I went with `WEBBACKEND` as you asked since making builds with both is probably some time out.

ccordoba12 commented:
Thanks @mingwandroid!

@msarahan, I think this one is ready to be merged!

mingwandroid commented:
I think it's nearly there. We need to resolve what to do about `conda-forge/obviousci`. Here I used it to setup the right Visual Studio environment, in particular the Windows 7.1 SDK.

Also, the commits marked as `[WIP]` should probably be modified to not say that. I don't think they are particularly `WIP`.

mingwandroid committed:
Darwin: Do not skip script module

Qt Creator needs it, and RStudio might too.

mingwandroid committed:
Add quotes to ${RECIPE_DIR}

.. As it could have spaces in it, for example if staged-recipes is
stored on a VMWare shared folder, by default you'll end up in a
subdirectory of::

"/Volumes/VMware Shared Folders"

mingwandroid committed:
Do not remove all static libraries

mingwandroid committed:
Darwin: WebEngine compatibility with OS X 10.7

This is disabled for now as there's some more work to do.

ccordoba12 committed:
Add selectors for last added patches

mingwandroid committed:
Add, but do not apply more osx patches

ccordoba12 commented:
@msarahan, please also close this one.
